common/src/main/java/com/microsoft/identity/common/internal/authorities/AADTestAuthority.java

```diff
@@ -0,0 +1,59 @@
+package com.microsoft.identity.common.internal.authorities;
+
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.providers.microsoft.microsoftsts.MicrosoftStsOAuth2Configuration;
+import com.microsoft.identity.common.internal.providers.oauth2.OAuth2Strategy;
+import com.microsoft.identity.common.internal.providers.ropc.ResourceOwnerPasswordCredentialsTestStrategy;
+
+import java.util.Map;
+
+public class AADTestAuthority extends AzureActiveDirectoryAuthority {
+
+    private static transient final String TAG = AADTestAuthority.class.getSimpleName();
+
+    public AADTestAuthority() {
+        // using organizations audience as common does not support ropc
+        super(new AnyOrganizationalAccount());
+    }
+
+    @Override
+    public OAuth2Strategy createOAuth2Strategy() {
+        final String methodName = ":createOAuth2Strategy";
```

Comment:
might be worth creating a new function that takes in config and returns a strategy in AzureActiveDirectoryAuthority, and override that instead.

i.e. 
protected OAuth2Strategy returnStrategyWithConfig(MicrosoftStsOAuth2Configuration config){
      // returns MicrosoftStsOAuth2Strategy or ROPCStrategy.
}

Reply:
Here and everywhere we're duplicating codes. Less code less maintenance :D

Reply:
Yeah that sounds good, will do that